core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/model/Extension.java

```diff
@@ -40,7 +40,10 @@ public boolean equals(Object obj) {
     @Deprecated(forRemoval = true)
     @JsonIgnore
     public boolean isMixedModule() {
-        return "io.quarkus".equals(groupId) && ("quarkus-core".equals(artifactId) || "quarkus-messaging".equals(artifactId));
+        return "io.quarkus".equals(groupId) && ("quarkus-core".equals(artifactId) ||
+                "quarkus-vertx-http".equals(artifactId) ||
```

Comment:
Could you open issues for all these extensions to ask them to switch to the new interface-based model? 

(Damned, I'm involved in all 3 :-))

Reply:
Sure, but I'm happy to do that.

I have all the vertx-http work already done, but I haven't pushed it yet because many extensions rely on the config object to obtain `quarkus.http.port`, and this will be a big breaking change.

To move that forward, I think we need to agree on this random port implementation, and then come up with a specific API (that I've intended to proposed separately), to retrieve useful things. Something like:

```java
class QuarkusRuntime {
  static Integer httpPort()
  ...
}
```

Ideally, we should offer this API before the vert-x config changes so that users who have to move can use the new API instead.